server/src/main/java/org/opensearch/cluster/metadata/Context.java

```diff
@@ -0,0 +1,110 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.cluster.metadata;
+
+import org.opensearch.cluster.AbstractDiffable;
+import org.opensearch.common.annotation.ExperimentalApi;
+import org.opensearch.core.ParseField;
+import org.opensearch.core.common.io.stream.StreamInput;
+import org.opensearch.core.common.io.stream.StreamOutput;
+import org.opensearch.core.xcontent.ConstructingObjectParser;
+import org.opensearch.core.xcontent.ToXContentObject;
+import org.opensearch.core.xcontent.XContentBuilder;
+import org.opensearch.core.xcontent.XContentParser;
+
+import java.io.IOException;
+import java.util.Map;
+
+/**
+ * Class encapsulating the context metadata associated with an index template/index.
+ */
+@ExperimentalApi
+public class Context extends AbstractDiffable<ComposableIndexTemplate> implements ToXContentObject {
+
+    private static final ParseField NAME = new ParseField("name");
+    private static final ParseField VERSION = new ParseField("version");
+    private static final ParseField PARAMS = new ParseField("params");
+
+    public static final String LATEST_VERSION = "_latest";
```

Comment:
shouldnt this be actual version like 1?

Reply:
This is used when no version is provided by user in context, hence latest